### PR TITLE
fix windows glob issue

### PR DESCRIPTION
**Pull request by yxxhero**
https://github.com/helmfile/helmfile/issues/1560

### CODE REVIEW

**Comment by xiaomudk** [comment]
@yxxhero @mumoshu  please review

File: pkg/state/storage.go [hunk 1]
```diff
@@ -136,18 +136,23 @@ func (st *Storage) normalizePath(path string) string {
...
-func (st *Storage) JoinBase(relPath, GOOS string) string {
+func (st *Storage) JoinBase(relPath string) string {
 	path := filepath.Join(st.basePath, relPath)
-	if GOOS == "windows" {
-		return strings.ReplaceAll(path, "\\", "\\\\")
-	}
 	return path
 }
+
+func (st *Storage) normalizeSetFilePath(path, goos string) string {
```
Comment (yxxhero) [inline pkg/state/storage.go:152]: why do need goos args? we can get it from `runtime.GOOS`?

File: pkg/state/storage_test.go [hunk 2]
```diff
@@ -206,9 +203,76 @@ func TestJoinBase(t *testing.T) {
...
-			if got := storageIns.JoinBase(tt.path, tt.goos); got != tt.want {
+			if got := storageIns.JoinBase(tt.path); got != tt.want {
 				t.Errorf("JoinBase() = %v, want %v", got, tt.want)
 			}
 		})
 	}
 }
+
+func TestNormalizeSetFilePath(t *testing.T) {
+	// 创建一个测试用的 Storage 实例
```
Comment (yxxhero) [inline pkg/state/storage_test.go:214]: please use english.

**Comment by yxxhero** [comment]
@xiaomudk please fix CI. thanks so much.

File: pkg/state/storage.go (hunk 1 above)
Comment (xiaomudk) [reply pkg/state/storage.go:151]: To make it easier to write unit tests

**Comment by yxxhero** [comment]
LGTM

**Comment by yxxhero** [comment]
@xiaomudk

**Review by xiaomudk — APPROVED** [review]

**Pull request merged**